packages/button/src/clear-button.ts

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { CSSResultArray, TemplateResult, html, property } from 'lit-element';
+import { ButtonBase } from './button-base.js';
+import buttonStyles from './clear-button.css.js';
+import crossMediumStyles from '@spectrum-web-components/icon/lib/spectrum-icon-cross-medium.css.js';
+
+export class ClearButton extends ButtonBase {
```

Comment:
is there a reason you added this rather than making an `over-background` variant of `<sp-button>`?

Reply:
`<sp-button variant="overBackground">` already exists, but with different delivery. This is specifically the "Clear Button" sub-pattern and it's `variant="overBackground"` alterations, which the CSS outline as being unique: https://opensource.adobe.com/spectrum-css/components/clearbutton/

I struggle with how our library should support these sub-variants. Should they be new elements, should we pack the old elements with more attributes/etc., this seems like the right way being this ties content and style delivery into one place. Thoughts?

Reply:
Oh -- took me a while to realize that the "Clear Button" is really just a *specific* icon button with the x. (I was thinking Clear had to do with visibility...). It seems a little weird to me to create a custom button for this rather than just creating the icon button inline with the right icon. Just an observation though - no need to hold anything up here.